ocaml/typing/ctype.ml

```diff
@@ -2104,8 +2104,8 @@ let rec estimate_type_jkind env ty =
   end
   | Tvariant row ->
       if tvariant_not_immediate row
-      then Jkind (Type.Primitive.value ~why:Polymorphic_variant)
-      else Jkind (Type.Primitive.immediate ~why:Immediate_polymorphic_variant)
+      then Jkind (Type.Primitive.value ~why:Polymorphic_variant |> Jkind.of_type_jkind)
```

Comment:
In `jkind.ml`, I think it would be worth adding re-exporting the contents of `Type.Primitive` in `Primitive` but wrapped in `Jkind.of_type_jkind`.

Reply:
It was done this way, but I reverted everything after @goldfirere's reviews: https://github.com/ocaml-flambda/flambda-backend/pull/2776#discussion_r1680929984

Reply:
(Discussing this further over on that PR)

Reply:
If anything, I believe that due to the length of the rebase chain at this point I'd prefer to keep this on tab as a future refactor rather than acting on this right now. I think it's a change that won't get more difficult to make later (it just requires going through existing uses and adding promotions).

Reply:
The length of the rebase chain is not a good reason to avoid making changes to a PR that is in review. Now that you've got things set up properly, I think you should be able to merge changes from early PRs into later PRs with a single button press. You'll also only need to do it once at the point where the early PR is approved for merging.

Reply:
That's true. I was more thinking about merge conflicts than the rebase itself.